hazelcast/src/main/java/com/hazelcast/internal/serialization/InternalSerializationService.java

```diff
@@ -75,7 +75,11 @@ public interface InternalSerializationService extends SerializationService, Disp
 
     void writeObject(ObjectDataOutput out, Object obj);
 
-    <T> T readObject(ObjectDataInput in);
+    default <T> T readObject(ObjectDataInput in) {
+        return readObject(in, false);
+    }
+
+    <T> T readObject(ObjectDataInput in, boolean useBigEndianForReadingTypeId);
```

Comment:
I would suggest renaming this(maybe to readTopLevelObject) or documenting the differences between this and other `readObject` method. I believe the initial purpose of `readObject` was to just read nested objects, and it would be good to clearly mark their differences and use cases

Reply:
https://github.com/hazelcast/hazelcast/pull/22941/commits/a75cb588d4ca69762a389092469379900023393d